crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -268,11 +270,25 @@ fn get_mocked_function_cheat_status<'a>(
     if call.call_type == CallType::Delegate {
         return None;
     }
-
-    cheatnet_state
+    match cheatnet_state
         .mocked_functions
         .get_mut(&call.storage_address)
-        .and_then(|contract_functions| contract_functions.get_mut(&call.entry_point_selector))
+    {
+        None => None,
+        Some(contract_functions) => {
+            let call_data_hash = poseidon_hash_many(call.calldata.0.iter());
+            let key = (call.entry_point_selector, call_data_hash);
+            let key_zero = (call.entry_point_selector, Felt::zero());
+
+            match contract_functions.get(&key) {
+                Some(CheatStatus::Cheated(_, CheatSpan::TargetCalls(0))) => {
```

Comment:
After calling `get_mocked_function_cheat_status`, we call `.decrement_cheat_span` on the returned value. It changes the CheatSpan to `Uncheated` after it decreases to `0`. 

I don't think there can be a case where `TargetCalls` is equal to 0 unless explicitly set to this value somewhere.

Reply:
I assume this logic match arm was added to handle the case where entrypoint for specific calldata is no longer cheated but for any calldata still is, but it should already be handled by `_` case anyway.

Reply:
Turns out technically it could be set to `0` but it shouldn't be allowed.

Created #2927

Reply:
I'm not sure to understand what should I do here: remove my workaround?

Reply:
You don't have to match for the case where cheat span is `CheatSpan::TargetCalls(0)`. Technically, user could have somehow created it like this (as explained in #2927), but during normal operation it will either be `CheatSpan::TargetCalls(some value > 0)` or `CheatStatus::Uncheated`.

Reply:
See this https://github.com/foundry-rs/starknet-foundry/blob/f2e7d0d4077a654992914a688575a85bbe57b9a9/crates/cheatnet/src/state.rs#L141

Reply:
Workaround removed